### ISSUE TITLE
Only fraction of paths defined by Icon Theme Specification gets considered

### ISSUE THREAD

**Issue by pmattern** [lxde/lxqt#976]
Icons located e. g. in `/usr/share/pixmaps/` are not displayed in the list of proposals that displays upon filing text in runner's form
![lxqt-runner_missing-icons-in-pixmaps](https://cloud.githubusercontent.com/assets/9860801/13574829/336a183e-e486-11e5-8af8-4142aa363b9f.png)

If I'm not mistaken it's the same problem as discussed regarding PCManFM-Qt in lxde/pcmanfm-qt#198.

**Comment by palinek** [comment]
@pmattern Please, test with the lxde/libqtxdg#77

**Comment by luis-pereira** [comment]
I'm able to see the icon without https://github.com/lxde/libqtxdg/pull/77
Have to dig in it.

**Comment by palinek** [comment]
@luis-pereira test with some scalable (svg) icon from /usr/share/pixmaps

**Comment by luis-pereira** [comment]
@pmattern Which icon theme are you using ?

**Comment by palinek** [comment]
I've put another fix to lxde/libqtxdg#77 for the non-scalable images found in directories w/o defined size.

**Comment by luis-pereira** [comment]
@pmatterm: @palinek fixes it. But there is deeper issue here:
An "modern" app shouldn't install icons to `/usr/share/pixmaps`. It should install it's icons to the hicolor theme.
From the standard:
> In order to have a place for third party applications to install their icons there should always exist a theme called "hicolor" [1]. The data for the hicolor theme is available for download at: http://www.freedesktop.org/software/icon-theme/. Implementations are required to look in the "hicolor" theme if an icon was not found in the current theme.

It's also an upstream/distribution issue.

**Comment by pmattern** [comment]
Sorry for not being able to participate earlier. Apparently there's quite some misunderstanding.

This issue has nothing to do with faulty scaling which may of course very well be a problem on its own.
Rather it is about genuine icons of e. g. applications the names of which are *not* covered by the [Icon Naming Specification](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html) and which are placed at locations covered by the [Icon Theme Specification](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html) yet independent from hicolor-icon-theme.

Steps to reproduce:
* Make sure a theme is used that does *not* comprise icons of xterm, e. g. oxygen.
* Check which icon name is used by xterm's desktop entry file and where it's located
```
$ grep -E '^Icon' /usr/share/applications/xterm.desktop 
Icon=xterm-color_48x48

$ locate xterm-color_48x48 
/usr/share/pixmaps/xterm-color_48x48.xpm
```
* Open runner's GUI and file "xterm". The icon will not be found as depicted in first comment.
* Move file `xterm-color_48x48.xpm` to `/usr/share/icons/hicolor/48x48/apps`. Restart the lxqt-runner binary ("LXQt Module" from `lxqt-config-session`) or start another session.
* Open runner's GUI and file "xterm". The icon will now appear.

Of course directories like `/usr/share/pixmaps` are more and more deprecated in favour of icon theme hicolor. But they still belong to the Icon Theme Specification and should hence be considered.
So all in all I still think this issue is just another manifestation of the the Qt problem mentioned int the first comment.

**Comment by palinek** [comment]
@pmattern You are right. But did you test with the fixed libqtxdg? These fixes aren't about scaling, but about showing the icon at all (in particular cases the icon engine did "pretend" that the icon provides image with size 0x0).

**Comment by luis-pereira** [comment]
@pmattern Sure it's still part of the specification, but it should be used only be legacy applications. An app like xterm should use the `hicolor` theme.

**Comment by pmattern** [comment]
@palinek 
You were right. Problem has nothing to do with the one of PCManFM-Qt stated in the first comment (which persists as of lxde/pcmanfm-qt@7ef9555, btw.) but got fixed in lxde/libqtxdg@b85f5fd (lxde/libqtxdg@18a4abf itself didn't compile on Arch Linux).

@luis-pereira 
While I personally agree icons are better of in theme hicolor the number of applications which keep using `/usr/share/pixmaps` is still rather large so I think it's pretty good we've got this working now.
